### ISSUE TITLE
Non-Explicit Version of Songs Playing

### ISSUE THREAD

**Issue by GhoulBoii** [KRTirtho/spotube#858]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When playing a spotify playlist which has explicit songs, spotube plays the non-explicit version of some of the songs. 

### Expected Behavior

The app should play the explicit version if the explicit version of the song is in the playlist.

### Steps to reproduce

1. Open Spotube
2. Play [Coffin by Lil Yachty](https://open.spotify.com/track/5vDNippoMr52KpXBO9b9KQ?si=c7169f102a0b4b5e)
3. The non-explicit version of the song is played

### Operating System

Android

### Spotube version

v3.2.0

### Installation source

GitHub Releases (Binary)

### Additional information

Occurs with the source being set as youtube and piped.

**Comment by csoftware-arigpt** [comment]
YouTube gives non-explicit version if you're not logged in on YouTube.

**Comment by GhoulBoii** [comment]
> YouTube gives non-explicit version if you're not logged in on YouTube.

Not true. Youtube is giving the explicit version as well when I check the alternative track sources. Piped+Youtube Music also gives the explicit version but it is the second result and not the first result so it does not play it
